jq.pyx

```diff
@@ -199,13 +257,7 @@ cdef class _ProgramWithInput(object):
         return _ResultIterator(self._jq_state_pool, self._bytes_input)
 
     def text(self):
-        iterator = self._make_iterator()
-        results = []
-        while True:
-            try:
-                results.append(iterator._next_string())
-            except StopIteration:
-                return "\n".join(results)
+        return "\n".join(json.dumps(v) for v in self)
```

Comment:
Is using `_jv_to_python` and `json.dumps` faster than using `jv_dump_string`?

Reply:
I haven't tested it, but very likely not. However, it's simpler. Tell me if you'd prefer performance over simplicity here. Provided there's performance benefit.

Reply:
I think it'd be useful to know what the performance implications are. I (perhaps naively!) think that it shouldn't introduce too much complexity -- instead of just `_next_string`, we could have something like `_next_result` which returns a `jv`, and then the two different methods (`text` and `__next__`) can handle the result appropriately.

Reply:
I'll try to throw together a little test for this, but meanwhile I pushed the change and it took me 25 lines:
```diff
diff --git a/jq.pyx b/jq.pyx
index f5642fe..54f024a 100644
--- a/jq.pyx
+++ b/jq.pyx
@@ -254,7 +254,13 @@ cdef class _ProgramWithInput(object):
         return _ResultIterator(self._jq_state_pool, self._bytes_input)
 
     def text(self):
-        return "\n".join(json.dumps(v) for v in self)
+        iterator = self._make_iterator()
+        results = []
+        while True:
+            try:
+                results.append(iterator._next_string())
+            except StopIteration:
+                return "\n".join(results)
 
     def all(self):
         return list(self)
@@ -288,7 +294,25 @@ cdef class _ResultIterator(object):
         return self
 
     def __next__(self):
+        cdef jv value
+        self._next_jv(&value)
+        return _jv_to_python(value)
+
+    cdef unicode _next_string(self):
         cdef int dumpopts = 0
+        cdef jv value
+        cdef jv dump
+
+        self._next_jv(&value)
+        dump = jv_dump_string(value, dumpopts)
+        try:
+            string = jv_string_value(dump).decode("utf8")
+        finally:
+            jv_free(dump)
+
+        return string
+
+    cdef int _next_jv(self, jv *presult) except 1:
         while True:
             if not self._ready:
                 self._ready_next_input()
@@ -296,7 +320,8 @@ cdef class _ResultIterator(object):
 
             result = jq_next(self._jq)
             if jv_is_valid(result):
-                return _jv_to_python(result)
+                presult[0] = result
+                return 0
             elif jv_invalid_has_msg(jv_copy(result)):
                 error_message = jv_invalid_get_msg(result)
                 message = jv_string_value(error_message).decode("utf8")
```